mono/mini/aot-compiler.c

```diff
@@ -106,6 +106,12 @@ get_method_nofail (MonoClass *klass, const char *method_name, int param_count, i
 #define EMIT_WIN32_UNWIND_INFO
 #endif
 
+#if defined(TARGET_ANDROID) || defined(__linux__)
+#define RODATA_REL_SECT ".data.rel.ro"
+#else
+#define RODATA_REL_SECT ".text"
+#endif
```

Comment:
Why limit this change only to Android or Linux? Wouldn't other target systems benefit from the lack of relocations in the code segment?

Reply:
On macOS the assembler bails out with `unknown directive` on `.data.rel.ro`.  Maybe there is something similar available but I didn't investigate. Likewise for Windows.

Reply:
Limiting it to android limits the amount of regressions which can be caused by this change.